feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsViewModel.kt

```diff
@@ -50,15 +62,311 @@ class SettingsViewModel @Inject constructor(
+            }
+        ) {
+            return FlipLensUiState.Disabled(
+                currentLensFacing = currentSettings.cameraLensFacing,
+                disabledRationale = setOf(DeviceUnsupportedRationale(LENS_SETTING_PREFIX))
+            )
+        }
+
+        // If multiple lens available, continue
+        val newLensFacing = if (currentSettings.cameraLensFacing == LensFacing.FRONT) {
+            LensFacing.BACK
+        } else {
+            LensFacing.FRONT
+        }
+        val constraintsRationale: MutableSet<DisabledRationale> = mutableSetOf()
+        val newLensConstraints = systemConstraints.perLensConstraints[newLensFacing]!!
+        // make sure all current settings wont break constraint when changing new default lens
+
+        // if new lens won't support fps
+        if (currentSettings.targetFrameRate != FPS_AUTO &&
+            !newLensConstraints.supportedFixedFrameRates
+                .contains(currentSettings.targetFrameRate)
+        ) {
+            constraintsRationale.add(
+                FpsUnsupportedRationale(
+                    LENS_SETTING_PREFIX,
+                    currentSettings.targetFrameRate
+                )
+            )
+        }
+
+        // if preview stabilization is currently on and the other lens won't support it
+        if (currentSettings.previewStabilization == Stabilization.ON) {
+            if (!newLensConstraints.supportedStabilizationModes.contains(
+                    SupportedStabilizationMode.ON
+                )
+            ) {
+                constraintsRationale.add(StabilizationUnsupportedRationale(LENS_SETTING_PREFIX))
+            }
+        }
+        // if video stabilization is currently on and the other lens won't support it
+        if (currentSettings.videoCaptureStabilization == Stabilization.ON) {
+            if (!newLensConstraints.supportedStabilizationModes
+                    .contains(SupportedStabilizationMode.HIGH_QUALITY)
+            ) {
+                constraintsRationale.add(StabilizationUnsupportedRationale(LENS_SETTING_PREFIX))
+            }
+        }
+
+        return if (constraintsRationale.isEmpty()) {
+            FlipLensUiState.Enabled(currentLensFacing = currentSettings.cameraLensFacing)
+        } else {
+            FlipLensUiState.Disabled(
+                currentLensFacing = currentSettings.cameraLensFacing,
+                constraintsRationale
+            )
+        }
+    }
+
+    private fun getFpsUiState(
+        systemConstraints: SystemConstraints,
+        cameraAppSettings: CameraAppSettings
+    ): FpsUiState {
+        val optionConstraintRationale: MutableMap<Int, SingleSelectableState> = mutableMapOf()
+
+        val deviceFrameRates: Set<Int> =
+            systemConstraints.perLensConstraints.values.fold(emptySet()) { union, constraints ->
+                union + constraints.supportedFixedFrameRates
+            }
```

Comment:
Since we now seem to be disabling settings based on the lens, I don't think we should do a union of all the lenses anymore. We should get the `deviceFrameRates` from the current `cameraLensFacing`.